modules/navier_stokes/include/fvbcs/WCNSFVEnergyFluxBC.h

```diff
@@ -47,4 +47,12 @@ class WCNSFVEnergyFluxBC : public FVFluxBC, public INSFVFlowBC
 
   /// Fluid specific heat capacity functor
   const Moose::Functor<ADReal> * const _cp;
+
+  /// The direction in which the flow is entering/leaving the domain. This is mainly used for cases
+  /// when the orientation of the face cannot be established (boundary on an internal face) or when the flow is
+  /// entering with an angle compared to the boundary surface.
+  const Point _direction;
+
+  /// Flag to store if the flow direction is specified by the user
+  const bool _direction_specified_by_user;
```

Comment:
Maybe we should create a base class for these WCNSFV classes at some point, but you've already done a lot in this PR

Reply:
Just mentioning that because of the apparent duplication between these three headers

Reply:
I'll add the base class, wanna make this clean.